cmd/apiserver_receive_adapter/main.go

```diff
@@ -94,43 +84,33 @@ func main() {
 		logger.Fatalw("Error building cloud event client", zap.Error(err))
 	}
 
-	controllers := []*kncontroller.Impl{}
+	gvrcs := []apiserver.GVRC(nil)
 
-	// Create one controller per resource.
 	for i, apiVersion := range env.ApiVersion {
 		kind := env.Kind[i]
 		controlled := env.Controller[i]
 
-		obj := &duckv1alpha1.AddressableType{}
-
-		factory := duck.TypedInformerFactory{
-			Client:       client,
-			ResyncPeriod: time.Duration(10), // TODO
-			StopChannel:  stopCh,
-			Type:         obj,
-		}
-
 		gv, err := schema.ParseGroupVersion(apiVersion)
 		if err != nil {
 			logger.Fatalw("Error parsing APIVersion", zap.Error(err))
 		}
-
-		gvk := schema.GroupVersionKind{Kind: kind, Group: gv.Group, Version: gv.Version}
-
 		// This is really bad.
-		gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+		gvr, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{Kind: kind, Group: gv.Group, Version: gv.Version})
+		gvrcs = append(gvrcs, apiserver.GVRC{
+			GVR:        gvr,
+			Controller: controlled,
+		})
+	}
 
-		// Get and start the informer for gvr
-		logger.Infof("Starting informer for %v", gvk)
-		informer, lister, err := factory.Get(gvr)
-		if err != nil {
-			logger.Fatalw("Error starting informer", zap.Error(err))
-		}
-		controllers = append(controllers, apiserver.NewController(opt, informer, lister, eventsClient, controlled))
+	opt := apiserver.Options{
+		Namespace: env.Namespace,
```

Comment:
my diff at least shows spaces between env. and Namespace and below. Just a bad diff?

Reply:
I see `Namespace: env.Namespace,`